rust/agents/validator/src/submit.rs

```diff
@@ -136,6 +136,10 @@ impl ValidatorSubmitter {
             self.submit_checkpoints_until_correctness_checkpoint(&mut tree, &latest_checkpoint)
                 .await?;
 
+            self.checkpoint_syncer
+                .write_latest_index(latest_checkpoint.index)
```

Comment:
wonder if we should have some logic like this https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/2839/files#diff-97b8300d4321afb2d386072aadf926cea1cc87f2e625e68bc427b25a176f1808L102-L109 to prevent a bunch of writes

Reply:
agreed

Reply:
not totally clear to me what we gain from this if we still have to do read IO?